### DISCUSSION TITLE
Enhance a class encounter error

### DISCUSSION THREAD

**Discussion by lingli0219** [apache/skywalking#8621] [Q&A] [answered]
I have encountered a error in logs of skywalking agent. I was trying to write a plugin for DB2. The error is like below:
![微信图片_](https://user-images.githubusercontent.com/31304902/156743525-45cc38aa-26e9-4204-a01b-d60e21a4f9bb.jpg)
I don't know how to deal with it. I can't see that the class aq has any special characteristics.
The error class aq implements DB2Statement interface. DB2Statement extends interface java.sql.Statement. The method executeLargeBatch is not implemented in both DB2Statement and aq and just use the default implementation in java.sql.Statement. The default implementation of the executeLargeBatch method in java.sql.Statement just throws an exception.
Please help me.

**Comment by wu-sheng** [comment, marked as answer]
It doesn't implement an interface, so? 

I can see from the logs, you are enhancing native methods? This is impossible, such as `notify`, `notifyAll`. I think you don't declare the instrumentation correctly. Also, you should know, DB2 is a commercial product owned by IBM, the open-source community should not involve too much. This is generally your private work unless you are going to open source in your repository.

**Reply by lingli0219** [reply]
Thank you very much for your answer. I did not mean to enhance native methods. I wll check my code next Monday. Maybe there is something wrong with my code.

**Reply by lingli0219** [reply]
I checked my code today. I can't see anything wrong. Here is my code
![微信图片_20220307151657](https://user-images.githubusercontent.com/31304902/156985361-fd480ee7-6b18-4251-af18-d17ca2672f3c.jpg)
:
Please give me a hand.

**Reply by wu-sheng** [reply]
I can't see the full error message about what is the illegalArgumentException. It should be found out by yourself locally.

**Reply by lingli0219** [reply]
I really can't figure it out all by myself. Please help me. Here are the top messages which I took from the full error message and which I consider as the important ones:
![微信图片_1](https://user-images.githubusercontent.com/31304902/156995871-aacf584a-457a-40a2-9946-5319925a441f.jpg)
![微信图片_2](https://user-images.githubusercontent.com/31304902/156995912-dd20fd2f-50e6-4466-abc8-ed4b040a370a.jpg)
![微信图片_3](https://user-images.githubusercontent.com/31304902/156995947-62d91690-bec9-497c-8299-5ab572003612.jpg)
![微信图片_4](https://user-images.githubusercontent.com/31304902/156995964-1eed571e-a5ca-4780-b0d5-18bfa5acefe8.jpg)
Please do help me. If you need the left messages from the full error message, I can supply. And I think maybe it's a bug.

**Reply by wu-sheng** [reply]
You provide a stack, I mean there is a `illegalArgumentException`, I want to see what is it saying?

I have said, you should know, one you are developing a private plugin, which is not accessible for me, two you are enhancing DB2, which is a commercial product. On both sides, this is 100% out of community scope. I just could provide some ideas but you should work on your own.

This is an open-source community, not service support.

**Reply by lingli0219** [reply]
I see. But I can't get help from anywhere else and I feel a little desperate. The illegalArgumentException is shown in my original post. Am I misunderstanding your idea again?

**Reply by wu-sheng** [reply]
> I can't get help from anywhere 

Sorry to hear that, but that is the reality. We must set up the boundary, open-source is making people work together. Once you are working for your company's private agenda, that is not our concern.
